portal/component-tests/test-assertions/pagination-assertions.js

```diff
@@ -0,0 +1,211 @@
+      wrapper.expectElement(`[data-cy="Page_${i}_listItem"]`).hasClass('govuk-!-margin-bottom-3');
+
+      wrapper.expectLink(`[data-cy="Page_${i}"]`).toLinkTo(`/testRoute/${i}?testQuery=test`, (i + 1).toString());
+      wrapper.expectElement(`[data-cy="Page_${i}"]`).hasClass('govuk-link');
+
+      if (i === currentPage) {
+        wrapper.expectElement(`[data-cy="Page_${i}_listItem"]`).hasClass('active');
+      } else {
+        wrapper.expectElement(`[data-cy="Page_${i}_listItem"]`).doesNotHaveClass('active');
+      }
+    }
+
+    wrapper.expectElement(`[data-cy="Page_${lastPage + 1}_listItem"]`).notToExist();
+    wrapper.expectLink(`[data-cy="Page_${lastPage + 1}"]`).notToExist();
+  });
+}
+
+function itShouldRenderEllipsisBeforeNextAndLastLinks(getWrapper) {
+  it('should render an ellipsis before the \'Next\' and \'Last\' links', () => {
+    const wrapper = getWrapper();
+
+    wrapper.expectText('[data-cy="nextLastEllipsis"]').toRead('...');
+    wrapper.expectElement('[data-cy="nextLastEllipsis"]').hasClass('govuk-body');
+    wrapper.expectElement('[data-cy="nextLastEllipsis"]').hasClass('govuk-!-margin-bottom-3');
+  });
+}
+
+function itShouldNotRenderEllipsisBeforeNextAndLastLinks(getWrapper) {
+  it('should NOT render an ellipsis before the \'Next\' and \'Last\' links', () => {
+    const wrapper = getWrapper();
+
+    wrapper.expectText('[data-cy="nextLastEllipsis"]').notToExist();
+  });
+}
+
+function itShouldRenderNextAndLastLinks({ getWrapper, currentPage, totalPages }) {
+  it('should render a \'Next\' link', () => {
+    const wrapper = getWrapper();
+
+    wrapper.expectElement('[data-cy="Next_listItem"]').hasClass('govuk-body');
+    wrapper.expectElement('[data-cy="Next_listItem"]').hasClass('govuk-!-margin-bottom-3');
+
+    wrapper.expectLink('[data-cy="Next"]').toLinkTo(`/testRoute/${currentPage + 1}?testQuery=test`, 'Next');
+    wrapper.expectElement('[data-cy="Next"]').hasClass('govuk-link');
+  });
+
+  it('should render a \'Last\' link', () => {
+    const wrapper = getWrapper();
+
+    wrapper.expectElement('[data-cy="Last_listItem"]').hasClass('govuk-body');
+    wrapper.expectElement('[data-cy="Last_listItem"]').hasClass('govuk-!-margin-bottom-3');
+
+    wrapper.expectLink('[data-cy="Last"]').toLinkTo(`/testRoute/${totalPages - 1}?testQuery=test`, 'Last');
+    wrapper.expectElement('[data-cy="Last"]').hasClass('govuk-link');
+  });
+}
+
+function itShouldNotRenderNextAndLastLinks(getWrapper) {
+  it('should NOT render a \'Next\' link', () => {
+    const wrapper = getWrapper();
+
+    wrapper.expectElement('[data-cy="Next_listItem"]').notToExist();
+
+    wrapper.expectLink('[data-cy="Next"]').notToExist();
+  });
+
+  it('should NOT render a \'Last\' link', () => {
+    const wrapper = getWrapper();
+
+    wrapper.expectElement('[data-cy="Last_listItem"]').notToExist();
+
+    wrapper.expectLink('[data-cy="Last"]').notToExist();
+  });
+}
+
```

Comment:
Nice, maybe could rename the file so it's `test-assertions/pagination`? Since the parent directory already has "assertions"

What do you think about splitting these up more so each function is in it's own file in a `pagination` directory, then this file can be a simple `index` with imports/exports?

Reply:
`What do you think about splitting these up more so each function is in it's own file in a pagination directory, then this file can be a simple index with imports/exports?`

I've done this :)